kernel/cond.c

```diff
@@ -320,3 +284,12 @@ int myst_cond_requeue(
 
     return 0;
 }
+
+int myst_cond_timedwait(
+    myst_cond_t* c,
+    myst_mutex_t* mutex,
+    const struct timespec* timeout)
+{
+    myst_signal_process(myst_thread_self());
```

Comment:
why not do the same while loop that we have in mutex?
If not then we dont need this processing of signals (which should have been after the timed wait anyway)